src/swap/types.ts

```diff
@@ -67,18 +97,48 @@ export type SwapError = {
   error: string; // The error message
 };
 
+/**
+ * Note: exported as public Type
+ */
+export type SwapParams = GetQuoteParams | GetSwapParams;
+
+export type SwapAPIParams = GetQuoteAPIParams | GetSwapAPIParams;
+
+/**
+ * Note: exported as public Type
+ */
+export interface Transaction {
+  transaction: TransactionData;
+
+  withParams(params: TransactionParams): TransactionData;
+}
+
+/**
+ * Note: exported as public Type
+ */
+export type TransactionData = {
```

Comment:
Same comment here on being able to use a `viem` type

Reply:
Which one are you thinking about?